### DISCUSSION TITLE
Configuring TLS email

### DISCUSSION THREAD

**Discussion by j-kia** [directus/directus#11236] [Archive]
Email connection fails when I'm using email service

### To Reproduce
I want to use TLS so set SMTP Secure to true (I also tried port 465)
```
EMAIL_TRANSPORT: 'smtp'
EMAIL_FROM: 'kurs@axelina.com'
EMAIL_SMTP_HOST: 'smtp-relay.gmail.com'
EMAIL_SMTP_PORT: 25
EMAIL_SMTP_USER: ***********
EMAIL_SMTP_PASSWORD: *********
EMAIL_SMTP_SECURE: 'true'
```

### Errors Shown

Error when EMAIL_SMTP_SECURE is true
```
[Error: 139748882389808:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:
] {
  library: 'SSL routines',
  function: 'ssl3_get_record',
  reason: 'wrong version number',
  code: 'ESOCKET',
  command: 'CONN'
}
10:00:51 🚨 139748882389808:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:
 
Error: 139748882389808:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:

10:00:51 ✨ request errored POST 500 /email/admin 166ms
10:00:51 ⚠️  Email connection failed:
10:00:51 ⚠️  139748882389808:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:
```

**Comment by jamescammarano** [comment]
I followed instructions also using a google email. Here is my config:
```
EMAIL_FROM="<MY_OTHER_EMAIL>@gmail.com"
EMAIL_TRANSPORT="smtp"

# ## Email (SMTP Transport)
# EMAIL_SMTP_POOL=true
EMAIL_SMTP_HOST="smtp.gmail.com"
EMAIL_SMTP_PORT=465
# EMAIL_SMTP_IGNORE_TLS=false
EMAIL_SMTP_USER="<MY_EMAIL>@gmail.com"
EMAIL_SMTP_PASSWORD="<APP_SPECIFIC_PASSWORD>"
EMAIL_SMTP_SECURE=true # Use TLS
```
I tested the mailing ability through an @mention in a comment on an item.

It was able to send and I was able to receive it. I did have to set an app-specific password through google but that didn't give me the same error. 
https://support.google.com/mail/?p=InvalidSecondFactor 

With port 25 I got:
> [INTERNAL_SERVER_ERROR] connect ETIMEDOUT 209.85.232.109:25

Which was also not your error.


Is there a specific place you were using the email system?

**Comment by j-kia** [comment]
> Is there a specific place you were using the email system?

I get this error all the time wherever i want to use the email service. When sending invitation mail with directus app or using MailsService in custom endpoints to send email with template

**Comment by rijkvanzanten** [comment]
@j-kia Try setting `EMAIL_SMTP_SECURE` to false. When `false`, nodemailer will automatically decide whether or not to use SSL based on the host used

**Comment by j-kia** [comment]
> Try setting `EMAIL_SMTP_SECURE` to false. When `false`, nodemailer will automatically decide whether or not to use SSL based on the host used

I set it to false and get `Connection timeout` error

**Comment by j-kia** [comment]
I updated **openssl** and now it doesn't give me the `wrong version number` error, Now i just have `Connection timeout` error no matter setting `EMAIL_SMTP_SECURE` to `true` or `false` .

**Comment by jamescammarano** [comment]
Did you change your port back to 465? I got that error on port 25.

**Comment by j-kia** [comment]
@jaycammarano yes you're right, I was using port 25, I change it to 465 and now i get:

> Invalid login: 535-5.7.8 Username and Password not accepted.

I checked username and password and it was correct. it seems i should allow less secure app in my account settings. 
I will let you know if the problem is solved by doing this.

**Comment by jamescammarano** [comment]
> @jaycammarano yes you're right, I was using port 25, I change it to 465 and now i get:
> 
> > Invalid login: 535-5.7.8 Username and Password not accepted.
> 
> I checked username and password and it was correct. it seems i should allow less secure app in my account settings. I will let you know if the problem is solved by doing this.

Did you set an app-specific password with Google? I had to do that.

**Comment by rijkvanzanten** [comment]
It sounds to me like this is most likely a configuration issue, lets keep this discussion going in Discussions 👍🏻

**Comment by dataexcess** [comment]
Hi! I would love to accomplish this too! Where can I make the app specific password? 
I cannot seem to make it work on my google account

**Comment by dataexcess** [comment]
Ok for me it started working once I enabled 'less secure app access' in google 🙌

**Comment by mattjcowan** [comment]
Getting similar error with a very simple SMTP endpoint:

![image](https://user-images.githubusercontent.com/1571516/194149469-1d1a89b1-fc03-4350-8b5e-5bec5d223dbb.png)

Config settings (blah.com is fake domain, but same otherwise):

```
EMAIL_FROM="myuser@blah.com"
EMAIL_TRANSPORT="smtp"
EMAIL_SMTP_HOST="mail.blah.com"
EMAIL_SMTP_PORT=25
EMAIL_SMTP_SECURE=false
```

I verified that it works using `swaks` (a simple smtp test tool), it's able to send email with that host over port 25:

```
swaks --to myemail@mydomain.com --server mail.blah.com --port 25
```

Any ideas?

**Reply by mattjcowan** [reply]
Hmm, got it to work with double quotes around the port ... EMAIL_SMTP_PORT="25"

**Reply by azrikahar** [reply]
That would be odd indeed 🤔 would you be able to try verify whether removing the quotes again will make it not working anymore?

Personally had success with the following docker compose for testing purposes (without quotes around `EMAIL_SMTP_PORT`):

```yaml
services:
    directus:
        image: directus/directus:9.18.1
        ports:
            - 8055:8055
        environment:
            - KEY=directus-key
            - SECRET=directus-secret
            - ADMIN_EMAIL=admin@example.com
            - ADMIN_PASSWORD=password
            - EMAIL_TRANSPORT=smtp
            - EMAIL_SMTP_HOST=maildev
            - EMAIL_SMTP_PORT=1025

    maildev:
        image: maildev/maildev:latest
        ports:
            - 1080:1080 # Maildev Web UI
```

so not sure what could be the possible cause. With that said, I'd also recommend using a Nodemailer script if possible if you ever run into any other errors again, similar to what was done here: https://github.com/directus/directus/issues/15183#issuecomment-1224336417. That way it should be easier to pinpoint whether it's something to do with Nodemailer (what is being used when configuring SMTP email transport in Directus) or Directus side of things.

**Reply by mattjcowan** [reply]
Ok, so now I go back in and I can't reproduce and tried a couple setups on different machines (windows/linux, node 14 and 16), so not sure what happened. Thanks for the prompt to double check @azrikahar !! I'm still curious to what triggered that error then, I'll stay on the lookout.

**Reply by vishalchavda24** [reply]
I was using coolify server and added env variables in coolify project config and email were not working and after updating directly on docker compose its works. Thanks @azrikahar : )
